test/index.test-d.ts

```diff
@@ -52,87 +52,93 @@ const postgrest = new PostgrestClient<Database>(REST_URL)
   )
 
   {
-    const { data, error } = await postgrest.from('users').select('status').eq('status', 'ONLINE')
-    if (error) {
-      throw new Error(error.message)
+    const result = await postgrest.from('users').select('status').eq('status', 'ONLINE')
```

Comment:
Here for some reason (maybe TS upgrade) using this syntax:

```
const { data, error } = await ...

if (error) {
 throw XXX
}
// here data was still of type data[] | null
// because typescript seemed to fail to understand the branching
```

Changing to this syntax properly narrow down `result.data` as non nullable once we checked the error.